app/controllers/IncomeTaxHistoryController.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.implicits._
+import controllers.actions.ValidatePerson
+import controllers.auth.AuthAction
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.renderer.TemplateRenderer
+import uk.gov.hmrc.tai.config.ApplicationConfig
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.domain.Employment
+import uk.gov.hmrc.tai.service.{EmploymentService, PersonService, TaxAccountService, TaxCodeChangeService}
+import uk.gov.hmrc.tai.viewModels.incomeTaxHistory.{IncomeTaxHistoryViewModel, IncomeTaxYear}
+import views.html.incomeTaxHistory.IncomeTaxHistoryView
+import uk.gov.hmrc.tai.util.ViewModelHelper._
+import uk.gov.hmrc.play.views.helpers.MoneyPounds
+
+import java.time.LocalDate
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class IncomeTaxHistoryController @Inject()(
+  val config: ApplicationConfig,
+  personService: PersonService,
+  authenticate: AuthAction,
+  validatePerson: ValidatePerson,
+  incomeTaxHistoryView: IncomeTaxHistoryView,
+  mcc: MessagesControllerComponents,
+  taxAccountService: TaxAccountService,
+  taxCodeChangeService: TaxCodeChangeService,
+  employmentService: EmploymentService,
+  errorPagesHandler: ErrorPagesHandler
+)(implicit ec: ExecutionContext, templateRenderer: TemplateRenderer)
+    extends TaiBaseController(mcc) {
+
+  private def getIncomeTaxHistoryPreviousYearsSeq(nino: Nino, taxYear: TaxYear)(
+    implicit hc: HeaderCarrier): Future[Seq[IncomeTaxHistoryViewModel]] = {
+    //YourTaxCodeController.prevTaxCodes
+    val futureMaybeTaxCodes = taxCodeChangeService.lastTaxCodeRecordsInYearPerEmployment(nino, taxYear)
```

Comment:
The YourTaxCodeController uses two different endpoints to retrieve TaxCode, reason here:

https://jira.tools.tax.service.gov.uk/browse/ST-278